src/hermes_claude_agent_sdk/parity/packets.py

```
@@ -0,0 +1,441 @@
+    result = _obj(value, keys, "result")
+    if result["result_schema_version"] != "3.0.0": _fail("result schema version is invalid")
+    for key in ("freeze_sha256", "contract_sha256", "catalog_sha256", "source_map_sha256", "fixture_manifest_sha256"): _sha(result[key], f"result.{key}")
+    if not isinstance(catalog, Mapping): _fail("catalog is required to bind SDK ledger and cells")
+    _safe(catalog)
+    candidate, inventory = _candidate(result["candidate"]), _inventory(result["inventory"]); _fail("candidate/inventory binding is invalid") if inventory["candidate_sha256"] != candidate["candidate_sha256"] else None
+    run = _obj(result["run"], {"run_id_sha256", "scenario_sha256", "mode", "scope_partition_id", "session_scope", "candidate_unchanged", "cell_count", "required_cell_count", "terminal_event_count", "resume_boundary"}, "result.run")
+    _sha(run["run_id_sha256"], "run_id_sha256"); _sha(run["scenario_sha256"], "scenario_sha256"); _enum(run["mode"], {"deterministic", "integration", "live"}, "run.mode"); _pref(run["scope_partition_id"], "PART-", "run.scope_partition_id"); _enum(run["session_scope"], SCOPES, "run.session_scope"); _bool(run["candidate_unchanged"], "run.candidate_unchanged")
+    if not run["candidate_unchanged"]: _fail("run candidate_unchanged must be true")
+    _int(run["cell_count"], "run.cell_count"); _int(run["required_cell_count"], "run.required_cell_count"); _int(run["terminal_event_count"], "run.terminal_event_count")
+    if run["resume_boundary"] is not None: _resume_boundary(run["resume_boundary"])
+    cells = _arr(result["cells"], "result.cells"); ids: list[str] = []; statuses: list[str] = []; records: list[tuple[Mapping[str, Any], list[Mapping[str, Any]]]] = []; boundaries: set[str] = set(); billing_safe = True; isolation_safe = True; resume_safe = True; catalog_caps = {cap["id"]: cap for cap in catalog.get("capabilities", [])}; catalog_parts = {part.get("id"): part for part in catalog.get("scope_partitions", []) if isinstance(part, Mapping)}; partition = catalog_parts.get(run["scope_partition_id"])
```

Ruff (0.16.3)

[error] Multiple statements on one line (semicolon)

(E702)